File: hcd-web/packages/ui/src/tokens/typography.ts
// Typography tokens ported from Swift Typography.swift

export const typography = {
  display: {
    fontSize: '2.25rem',   // 36px
    lineHeight: '2.5rem',
    fontWeight: '700',
    letterSpacing: '-0.025em',
  },
  heading1: {
    fontSize: '1.5rem',    // 24px
    lineHeight: '2rem',
    fontWeight: '600',
    letterSpacing: '-0.02em',
  },
  heading2: {
    fontSize: '1.25rem',   // 20px
    lineHeight: '1.75rem',
    fontWeight: '600',
    letterSpacing: '-0.01em',
  },
  heading3: {
    fontSize: '1.125rem',  // 18px
    lineHeight: '1.75rem',
    fontWeight: '600',
  },
  body: {
    fontSize: '1rem',      // 16px
    lineHeight: '1.5rem',
    fontWeight: '400',
  },
  bodyMedium: {
    fontSize: '1rem',
    lineHeight: '1.5rem',
    fontWeight: '500',
  },
  caption: {
    fontSize: '0.875rem',  // 14px
    lineHeight: '1.25rem',
    fontWeight: '400',
  },
  small: {
    fontSize: '0.75rem',   // 12px
    lineHeight: '1rem',
    fontWeight: '400',
  },
} as const;

// Tailwind class equivalents
export const typographyClasses = {
  display: 'text-4xl font-bold tracking-tight',
  heading1: 'text-2xl font-semibold tracking-tight',
  heading2: 'text-xl font-semibold',
  heading3: 'text-lg font-semibold',
  body: 'text-base',
  bodyMedium: 'text-base font-medium',
  caption: 'text-sm',
  small: 'text-xs',
} as const;

File: HCDInterviewCoach/DesignSystem/Typography.swift
import SwiftUI

/// Typography tokens with Dynamic Type support on iOS.
/// On macOS, fixed sizes are used (desktop viewing distance).
/// On iOS, text styles map to Dynamic Type categories for accessibility.
enum Typography {
    #if os(iOS)
    static let display = Font.largeTitle.weight(.bold)
    static let heading1 = Font.title.weight(.semibold)
    static let heading2 = Font.title3.weight(.semibold)
    static let heading3 = Font.headline
    static let body = Font.body
    static let bodyMedium = Font.body.weight(.medium)
    static let caption = Font.caption
    static let small = Font.caption2
    #else
    static let display = Font.system(size: 32, weight: .bold)
    static let heading1 = Font.system(size: 24, weight: .semibold)
    static let heading2 = Font.system(size: 18, weight: .semibold)
    static let heading3 = Font.system(size: 16, weight: .semibold)
    static let body = Font.system(size: 14, weight: .regular)
    static let bodyMedium = Font.system(size: 14, weight: .medium)
    static let caption = Font.system(size: 12, weight: .regular)
    static let small = Font.system(size: 10, weight: .regular)
    #endif
}
